### DISCUSSION TITLE
Unable to bump cask version

### DISCUSSION THREAD

**Discussion by eloyaldea** [Homebrew/discussions#539] [Casks]
Hi, I'm trying to bump the versions of a couple casks: `whatsapp` and `streamlabs-obs` with the following command:
`brew bump-cask-pr --version 2.2100.4 whatsapp` and `brew bump-cask-pr --version 0.27.1 streamlabs-obs`

The following output is presented:

> ==> Downloading https://web.whatsapp.com/desktop/mac/files/release-2.2100.4.zip
Already downloaded: /Users/XXX/Library/Caches/Homebrew/downloads/f7d65f82e1db664c6e3db9002e9cc530ca172328d5e3e2ca9d91e9aacf9da809--_nc_ht=scontent.fmad12-2.fna
Warning: Cannot verify integrity of 'f7d65f82e1db664c6e3db9002e9cc530ca172328d5e3e2ca9d91e9aacf9da809--_nc_ht=scontent.fmad12-2.fna'.
No checksum was provided for this resource.
For your reference, the checksum is:
  sha256 "ce3564d09d2c549a2899bbccae2ba7cf70efa0fbb27c216972c93857ad3dbfbc"
==> replace /version\s+["']2\.2100\.4["']/m with "version \"2.2100.4\""
==> replace /sha256\s+["']ce3564d09d2c549a2899bbccae2ba7cf70efa0fbb27c216972c93857ad3dbfbc["']/m with "sha256 \"ce3564d09d2c549a
audit for whatsapp: passed

> 1 file inspected, no offenses detected
fatal: A branch named 'bump-whatsapp-2.2100.4' already exists.
Error: Failure while executing; `git checkout --no-track -b bump-whatsapp-2.2100.4 origin/master` exited with 128.

My user has been replaced with XXX.

My Homebrew token has been setup with `gist` , `public_repo` and `workflow` permissions, as some homebrew output suggested:

> Error: Your HOMEBREW_GITHUB_API_TOKEN credentials do not have sufficient scope!
Scopes required: public_repo
Scopes present:  gist, workflow
Create a personal access token:
  https://github.com/settings/tokens/new?scopes=gist,public_repo&description=Homebrew
echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.zshrc

Also I have run `echo 'export HOMEBREW_GITHUB_API_TOKEN=your_token_here' >> ~/.zshrc` and `export HOMEBREW_GITHUB_API_TOKEN=your_token_here` and I don't know which is the best option. I manage to get the token working for a while and I'm able to submit PRs but after some time Homebrew tells me the token has expired, without me touching anything.

I don't see any PR opened for either of these casks so I don't know why it's not letting me create one.

Running `brew doctor` shows:

> Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

> Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/cask) checkout master

And running `git -C $(brew --repo homebrew/cask) checkout master` shows this output:

> error: Your local changes to the following files would be overwritten by checkout:
	Casks/adoptopenjdk.rb
	Casks/microsoft-office.rb
	Casks/portfolioperformance.rb
	Casks/streamlabs-obs.rb
	Casks/webcatalog.rb
Please commit your changes or stash them before you switch branches.
Aborting

I assume by the time someone is able to reply to this, both casks will have been updated by other users.

Thanks for the help!

**Comment by yannpub** [comment]
I am facing the exact same issue with keepassxc and mkvtoolnix.

**Reply by eloyaldea** [reply]
I have double checked everything, I'm new to using GitHub but I looked at guides and did everything I knew to understand it but I can't. Should I open an isssue rather than a discussion here?

**Reply by yannpub** [reply]
I ran again a brew update, brew upgrade, and the updates came. maybe there was some pending validations somewhere.

**Reply by eloyaldea** [reply]
Yes the updated have come because someone else made a PR for them, but I want to know why I couldn't do it in the first place.

**Comment by cesarcoatl** [comment]
I've had this happened to me before when something failed and the PR was not created. And in my case the branch only existed on my computer, so I did the following:

1. Go to `homebrew-cask` tap
    ```bash
    $ cd "$(brew --repository)"/Library/Taps/homebrew/homebrew-cask
    ```
2. List branches
    ```bash
    $ git branch
    ```
3. Delete local "offending" branch:
    ```bash
    $ git branch -D bump-whatsapp-2.2100.4
    ```
4. Retry `brew bump-cask-pr`

Additional tip: always provide `--sha256`. One way to determine that on a file you've downloaded to `~/Downloads` would be by running the following command:

```bash
$ shasum -a 256 ~/Downloads/foo.zip
```

**Reply by carlocab** [reply]
Another way to get the `sha256` is

    curl -L url/to/foo.tar.gz | sha256sum

Alternatively, you can just change the URLs in `foo.rb` then do

    brew fetch -s foo

`brew` will download the source tarballs and any declared resources and list their `sha256`s.

**Reply by eloyaldea** [reply]
Thanks for the help but I'm still having errors after deleting all my branches (I left the one called `master` as deleteing that didn't seem safe). `brew doctor` is saying this: 

> Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

> Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/cask) checkout master

And running `git -C $(brew --repo homebrew/cask) checkout master` tells me this:

> error: Your local changes to the following files would be overwritten by checkout:
	Casks/abricotine.rb
	Casks/adoptopenjdk.rb
	Casks/ajour.rb
	Casks/arq.rb
	Casks/blitz.rb
	Casks/boost-note.rb
	Casks/chef-workstation.rb
	Casks/cmake.rb
	Casks/cockatrice.rb
	Casks/coverload.rb
	Casks/cyberduck.rb
	Casks/docker.rb
	Casks/element.rb
	Casks/evernote.rb
	Casks/expandrive.rb
	Casks/fsnotes.rb
	Casks/github.rb
	Casks/hubstaff.rb
	Casks/irccloud.rb
	Casks/keepassxc.rb
	Casks/kotlin-native.rb
	Casks/lyricsx.rb
	Casks/macintoshjs.rb
	Casks/macupdater.rb
	Casks/metasploit.rb
	Casks/microsoft-office.rb
	Casks/musescore.rb
	Casks/netron.rb
	Casks/panoply.rb
	Casks/portfolioperformance.rb
	Casks/powershell.rb
	Casks/protonvpn.rb
	Casks/qownnotes.rb
	Casks/raven-reader.rb
	Casks/ray.rb
	Casks/routebuddy.rb
	Casks/rstudio.rb
	Casks/smartgit.rb
	Casks/streamlabs-obs.rb
	Casks/timeular.rb
	Casks/tinkerwell.rb
	Casks/unclack.rb
	Casks/webcatalog.rb
	Casks/wechat.rb
	Casks/whatsapp.rb
	audit_exceptions/github_prerelease_allowlist.json
Please commit your changes or stash them before you switch branches.
Aborting

I don't know what to do after this :/

**Reply by gromgit** [reply]
If you're sure you haven't touched any of those files (or you _did_ but don't want to keep the changes), you can force the checkout:
```
git -C $(brew --repo homebrew/cask) checkout -f master
```

**Reply by eloyaldea** [reply]
Thanks!!! I think it's working fine now, if it breaks again for some reason I'll go back to this thread and repeat the same steps. 😄

**Comment by bitmvr** [comment]
I followed the steps by @thecesrom and was finally successful in submitting my latest pull request.  I would amend the directions provided by Cesar to add step 2 to make sure you're not in a branch you're trying to delete in case you do not catch the error thrown by `git`.

1. Go to `homebrew-cask` tap
```
$ cd "$(brew --repository)"/Library/Taps/homebrew/homebrew-cask
```

2. Checkout master
```
$ git checkout master
```

3. List branches
```
$ git branch
```

4. Delete local "offending" branch:
```
$ git branch -D bump-whatsapp-2.2100.4
```

5.  Retry bump

```
brew bump-cask-pr --version <new_version> <outdated_cask>
```
